lib/campaign/CampaignState.cpp

```diff
@@ -455,6 +477,42 @@ std::set<CampaignScenarioID> Campaign::allScenarios() const
 	return result;
 }
 
+void Campaign::overrideCampaign(bool scenario)
```

Comment:
Perhaps split into two methods - override scenario & override campaign? Instead of bool parameter. Most of code is inside `if` block and `overrideCampaign(false)` is not exactly clear as to what it does